packages/next/next-server/server/render.tsx

```diff
@@ -411,6 +439,14 @@ export async function renderToHTML(
     </RouterContext.Provider>
   )
 
+  const invalidKeysMsg = (methodName: string, invalidKeys: string[]) => {
```

Comment:
I don't think it's clear this throws an error, which means TypeScript will misinterpret flow.

Can we make this only return the string instead?

`function getInvalidKeysMsg() { }`, then still construct error on call-site.

Reply:
Updated to only build and return a string. I also moved this out of render since it doesn't need to be in this scope